### PR TITLE
Blog: Deploying Windows FSx joined to an AD using AWS CDK

### PR DESCRIPTION
**Pull request by nidhi-s-2903**

### CODE REVIEW

**Comment by nidhi-s-2903** [comment]
Preview Link:
https://feature-blog-windows-fsx-aws-cdk--skyworkz.netlify.app/blog/windows-fsx-joined-to-ad-with-aws-cdk/

**Comment by bennycornelissen** [comment]
First impression: I'm impressed! Going to give it a more thorough read tomorrow morning because I'm tired now, but this looks pretty much good to go.

I would recommend squashing the commits before we merge though 😉

File: content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+author: Nidhi Sharma
+tags:
+  - AWS CDK
+  - Windows FSx
+  - Active Directory
+slug: windows-fsx-joined-to-ad-with-aws-cdk
+date: 2023-02-25
+---
+
+In a Windows Server setup, it is common practice to use Active Directory for server and user management. For this purpose, you can create an AWS managed Active Directory. Any EC2 instances or other resources that you would like to manage should join the AD upon their creation. In our setup, we required a common file storage for sharing resources across our Windows EC2 instances. For this purpose, we used AWS FSx for Windows.
```
Comment (LFSb) [inline content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md]: Suggestion: "In our setup, we required...." should probably be rewritten to "At a client, we had a setup that required...".

**Review by LFSb — DISMISSED** [review]
Very enjoyable read! Aside from a small suggestion in the first paragraph, I have no further additions. Thanks Nidhi :)

File: content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md (hunk 1 above)
Comment (LFSb) [inline content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md]: One more addition, maybe link to the https://aws.amazon.com/fsx/ page for people who don't already know what that is 😄

File: content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+### Roadblock coming ahead
+
+To join the AD, certain ports from the FSx's security group should be whitelisted in the Inbound rules of the AD's security group. Marked below in red in the diagram. This basically means, before creating your FSx, you must have:
+
+1. An Active Directory's ID
+2. The AD's security group ID
+3. The FSx's security group ID
+
+When you create an AD, a security group is automatically assigned to the AD and you cannot assign a different security group to it. In addition to that, you cannot figure out the security group ID after the AD is created as it is not returned by the CDK API via any property or method.
```
Comment (bennycornelissen) [inline content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md:28]: Tell me there's a bugreport for this.. 🙈

**Review by bennycornelissen — CHANGES_REQUESTED** [review]
I agree with the suggestions @LFSb made. It's a solid blogpost. I like it. And let's make sure to squash before merging

File: content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md (hunk 1 above)
Comment (nidhi-s-2903) [reply content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md]: Done

Comment (nidhi-s-2903) [reply content/blog/windows-fsx-joined-to-active-directory-using-aws-cdk.md]: Done

**Review by bennycornelissen — APPROVED** [review]
LGTM 🎉

**Pull request merged**